src/utils/__tests__/replacer.test.ts

```diff
@@ -53,16 +57,47 @@ describe('Replacer', () => {
     ).toEqual('missing :(');
   });
 
-  it('supports transformers', () => {
+  it('evaluates expressions', () => {
     const replacer = new Replacer<Dictionary<unknown>>(2);
-    replacer.addTransformer('dot', (id, value) => (Array.isArray(value) ? value.join('.') : String(value)));
 
-    const template = '{{path|dot}}';
+    const template = "#{{path[1] + '-' + path[2].toUpperCase()}}";
+
+    expect(
+      replacer.print(template, {
+        path: ['foo', 'bar', '/a'],
+      }),
+    ).toEqual('bar-/A');
+  });
+
+  it('supports custom functions', () => {
+    const replacer = new Replacer<Dictionary<unknown>>(2);
+    replacer.addFunction('printPath', function () {
+      const { path } = this;
+      return Array.isArray(path) ? path.join('.') : String(path);
+    });
+
+    const template = '#{{printPath()}}';
 
     expect(
       replacer.print(template, {
         path: ['foo', 'bar', '/a'],
       }),
     ).toEqual('foo.bar./a');
   });
+
+  it('handles and prints out exceptions thrown during evaluation', () => {
```

Comment:
```suggestion
  it('ignores and prints out exceptions thrown during evaluation', () => {
```
@nulltoken looks good? :grin: